### PR TITLE
[14][ADD] stock_move_auto_assign_auto_release - backport

### PR DESCRIPTION
**Pull request by TDu**
This module is being backported from v16, related PR:

* https://github.com/OCA/stock-logistics-warehouse/pull/1560

It depends on 
* https://github.com/OCA/stock-logistics-warehouse/pull/1625
* https://github.com/OCA/wms/pull/556

### CODE REVIEW

File: stock_move_auto_assign/tests/test_auto_assign.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
 from odoo.addons.queue_job.job import identity_exact
-from odoo.addons.queue_job.tests.common import mock_with_delay
+from odoo.addons.queue_job.tests.common import mock_with_delay, trap_jobs
```
Comment (mt-software-de) [inline stock_move_auto_assign/tests/test_auto_assign.py]: Could you please remove the deprecated mock_with_delay
I created a PR https://github.com/camptocamp/stock-logistics-warehouse/pull/2

**Review by mt-software-de — CHANGES_REQUESTED** [review]

**Comment by TDu** [comment]
~~The fix for the pre-commit is https://github.com/OCA/stock-logistics-warehouse/pull/1651~~

File: stock_move_auto_assign/tests/test_auto_assign.py (hunk 1 above)
Comment (TDu) [reply stock_move_auto_assign/tests/test_auto_assign.py]: I updated/fix the tests. no more with_delay

Comment (mt-software-de) [reply stock_move_auto_assign/tests/test_auto_assign.py]: thx

**Comment by TDu** [comment]
Squashed and removed resolved dev dependencies.

**Comment by TDu** [comment]
Test are failing because the last change on https://github.com/OCA/wms/tree/14.0/stock_available_to_promise_release are not visible in the build of this PR ?
The version for `stock_available_to_promise_release` is still the old one https://github.com/OCA/stock-logistics-warehouse/actions/runs/4416816728/jobs/7741637597#step:4:189 
Problem with the PyPi package ?

**Comment by TDu** [comment]
On PyPi the new 14.0.2.0.0 does not exist
```
—> pip index versions odoo14-addon-stock_available_to_promise_release
odoo14-addon-stock_available_to_promise_release (14.0.1.4.0)
Available versions: 14.0.1.4.0, 14.0.1.3.0, 14.0.1.2.0, 14.0.1.1.3, 14.0.1.1.2, 14.0.1.1.1, 14.0.1.1.0, 14.0.1.0.1

```
?

**Comment by TDu** [comment]
> On PyPi the new 14.0.2.0.0 does not exist
> 
> ```
> —> pip index versions odoo14-addon-stock_available_to_promise_release
> odoo14-addon-stock_available_to_promise_release (14.0.1.4.0)
> Available versions: 14.0.1.4.0, 14.0.1.3.0, 14.0.1.2.0, 14.0.1.1.3, 14.0.1.1.2, 14.0.1.1.1, 14.0.1.1.0, 14.0.1.0.1
> ```
> 
> ?

A fix was merged on `sale_available_to_promise_release` and there is PyPi package this time.

**Review by jbaudoux — APPROVED** [review]
LG

**Comment by jbaudoux** [comment]
@mt-software-de Can you update your review?

**Review by mt-software-de — APPROVED** [review]
LGTM

**Pull request merged**